packages/tools/src/tools/annotation/UltrasoundDirectionalTool.ts

```diff
@@ -0,0 +1,894 @@
+        const canvasTextBoxCoords = canvasCoordinates[1];
+
+        data.handles.textBox.worldPosition =
+          viewport.canvasToWorld(canvasTextBoxCoords);
+      }
+
+      const textBoxPosition = viewport.worldToCanvas(
+        data.handles.textBox.worldPosition
+      );
+
+      const textBoxUID = '1';
+      const boundingBox = drawLinkedTextBoxSvg(
+        svgDrawingHelper,
+        annotationUID,
+        textBoxUID,
+        textLines,
+        textBoxPosition,
+        canvasCoordinates,
+        {},
+        options
+      );
+
+      const { x: left, y: top, width, height } = boundingBox;
+
+      data.handles.textBox.worldBoundingBox = {
+        topLeft: viewport.canvasToWorld([left, top]),
+        topRight: viewport.canvasToWorld([left + width, top]),
+        bottomLeft: viewport.canvasToWorld([left, top + height]),
+        bottomRight: viewport.canvasToWorld([left + width, top + height]),
+      };
+    }
+
+    return renderStatus;
+  };
+
+  _calculateCachedStats(annotation, renderingEngine, enabledElement) {
+    const data = annotation.data;
+    const { viewportId, renderingEngineId } = enabledElement;
+
+    // Until we have all two anchors bail out
+    if (data.handles.points.length !== 2) {
+      return;
+    }
+
+    const { cachedStats } = data;
+    const targetIds = Object.keys(cachedStats);
+
+    for (let i = 0; i < targetIds.length; i++) {
+      const targetId = targetIds[i];
+
+      const image = this.getTargetIdImage(targetId, renderingEngine);
+
+      // If image does not exists for the targetId, skip. This can be due
+      // to various reasons such as if the target was a volumeViewport, and
+      // the volumeViewport has been decached in the meantime.
+      if (!image) {
+        continue;
+      }
+
+      const { imageData } = image;
+
+      const worldPos1 = data.handles.points[0];
+      const worldPos2 = data.handles.points[1];
+
+      const imageIndex1 = transformWorldToIndex(imageData, worldPos1);
+      const imageIndex2 = transformWorldToIndex(imageData, worldPos2);
+
+      const { values: values1, units: units1 } =
+        getCalibratedProbeUnitsAndValue(image, [imageIndex1]);
+      const { values: values2, units: units2 } =
+        getCalibratedProbeUnitsAndValue(image, [imageIndex2]);
+
+      let value;
+      let unit = 'px';
+      if (
+        units1[0] !== units2[0] ||
+        units1[1] !== units2[1] ||
+        (units1[0] === 'raw' && units2[0] === 'raw')
+      ) {
+        // if units are not the same, we cannot calculate the diff
+        // so we just report the px distance
+        value = distanceToPoint(worldPos1, worldPos2);
+      } else {
+        const canvasPoint1 = enabledElement.viewport.worldToCanvas(worldPos1);
+        const canvasPoint2 = enabledElement.viewport.worldToCanvas(worldPos2);
+
+        const canvasDeltaY = canvasPoint2[1] - canvasPoint1[1];
+        const canvasDeltaX = canvasPoint2[0] - canvasPoint1[0];
+
+        const orientation =
```

Comment:
Can you make this a boolean so that you only have the comparison once, eg
const orientationHorizontal = ....
It is a bit ugly to have the values all over the place for the comparison.